mesonbuild/interpreter/interpreter.py

```diff
@@ -3163,19 +3163,25 @@ def add_target(self, name: str, tobj: build.Target) -> None:
         # To permit an executable and a shared library to have the
         # same name, such as "foo.exe" and "libfoo.a".
         idname = tobj.get_id()
-        for t in self.build.targets.values():
-            if t.get_id() == idname:
-                raise InvalidCode(f'Tried to create target "{name}", but a target of that name already exists.')
-            if isinstance(tobj, build.Executable) and isinstance(t, build.Executable) and t.name == tobj.name:
-                FeatureNew.single_use('multiple executables with the same name but different suffixes',
-                                      '1.3.0', self.subproject, location=self.current_node)
+        subdir = tobj.get_subdir()
+        namedir = name + "," + subdir
```

Comment:
I think just using a tuple as a key would be better.

Reply:
Done